l1-contracts/contracts/bridge/L1SharedBridge.sol

```diff
@@ -0,0 +1,638 @@
+        l2BridgeAddress[ERA_CHAIN_ID] = ERA_ERC20_BRIDGE_ADDRESS;
+    }
+
+    /// @dev Initializes the l2Bridge address by governance for a specific chain.
+    function initializeChainGovernance(uint256 _chainId, address _l2BridgeAddress) external onlyOwner {
+        l2BridgeAddress[_chainId] = _l2BridgeAddress;
+    }
+
+    /// @notice Allows bridgehub to acquire mintValue for L1->L2 transactions.
+    /// @dev If the corresponding L2 transaction fails, refunds are issued to a refund recipient on L2.
+    function bridgehubDepositBaseToken(
+        uint256 _chainId,
+        address _prevMsgSender,
+        address _l1Token,
+        uint256 _amount
+    ) external payable virtual onlyBridgehubOrEra(_chainId) {
+        if (_l1Token == ETH_TOKEN_ADDRESS) {
+            require(msg.value == _amount, "L1SharedBridge: msg.value not equal to amount");
+        } else {
+            // The Bridgehub also checks this, but we want to be sure
+            require(msg.value == 0, "ShB m.v > 0 b d.it");
+
+            uint256 amount = _depositFunds(_prevMsgSender, IERC20(_l1Token), _amount); // note if _prevMsgSender is this contract, this will return 0. This does not happen.
+            require(amount == _amount, "3T"); // The token has non-standard transfer logic
+        }
+
+        if (!hyperbridgingEnabled[_chainId]) {
+            chainBalance[_chainId][_l1Token] += _amount;
+        }
+        // Note that we don't save the deposited amount, as this is for the base token, which gets sent to the refundRecipient if the tx fails
+    }
+
+    /// @dev Transfers tokens from the depositor address to the smart contract address.
+    /// @return The difference between the contract balance before and after the transferring of funds.
+    function _depositFunds(address _from, IERC20 _token, uint256 _amount) internal returns (uint256) {
+        uint256 balanceBefore = _token.balanceOf(address(this));
+        _token.safeTransferFrom(_from, address(this), _amount);
+        uint256 balanceAfter = _token.balanceOf(address(this));
+
+        return balanceAfter - balanceBefore;
+    }
+
+    /// @notice Initiates a deposit transaction within Bridgehub, used by `requestL2TransactionTwoBridges`.
+    function bridgehubDeposit(
+        uint256 _chainId,
+        address _prevMsgSender,
+        uint256, // l2Value, needed for Weth deposits in the future
+        bytes calldata _data
+    ) external payable override onlyBridgehub returns (L2TransactionRequestTwoBridgesInner memory request) {
+        require(l2BridgeAddress[_chainId] != address(0), "ShB l2 bridge not deployed");
+
+        (address _l1Token, uint256 _depositAmount, address _l2Receiver) = abi.decode(
+            _data,
+            (address, uint256, address)
+        );
+        require(_l1Token != l1WethAddress, "ShB: WETH deposit not supported");
+        require(bridgehub.baseToken(_chainId) != _l1Token, "ShB: baseToken deposit not supported");
```

Comment:
I know I'm being pesky... but hear me out...

There can be a token with multiple addresses. Check out [this implementation](https://github.com/d-xo/weird-erc20/blob/main/src/Proxied.sol).

Do what you may, with this info.

Reply:
It can be this and other crazy implementation, but let's just say we don't support such tokens :)